packages/js-lib/src/enum.util.ts

```diff
@@ -1,4 +1,34 @@
-import type { NumberEnum, StringEnum } from './types.js'
+import type { AnyObject, NumberEnum, StringEnum } from './types.js'
+
+export function isNumberEnum(en: AnyObject): en is NumberEnum {
+  /*
+   * enum Foo { A = 1, B = 2 }
+   * becomes
+   * { "1": "A", "2": "B", "A": 1, "B": 2}
+   */
+
+  return Object.values(en).every(value => {
+    const isValueString = isNaN(Number(value))
+    const isValueNumber = !isValueString
+
+    if (isValueNumber) {
+      return en[value] !== undefined
```

Comment:
I think it's ok to not support undefined values. It's an edge case and "bad Enum". I don't want to support "mixed value" enums. Instead only support "pure" enums, where all values are of the same type - either String or Number.

So, we can simplify the code a bit by having these assumptions 🙏

Reply:
This is not about undefined values.

In NumberEnum's when a value is string (instead of number),
it's because it's a "backlink" to the original numeric value.

So this condition is a check that to ensure that whenever we encounter a string value, it's only a backlink to a numeric value -> thus enforcing a NumberEnum.